subprojects/docs/src/docs/release/notes.md

```diff
@@ -10,6 +10,10 @@ Add-->
 
 Starting with this release, it is now possible to use SNAPSHOT plugin versions in the `plugins {}` and `pluginManagement {}` blocks.
 
+### Incremental Java compilation by default
+
+The incremental Java compiler is now enabled by default.
```

Comment:
We could enumerate the benefits from `javaPlugin.adoc` here so it's a bit more exciting. 😉